pride-hg-support/src/main/java/com/prezi/pride/vcs/hg/HgVcsSupport.java

```diff
@@ -0,0 +1,210 @@
+		}
+
+		ProcessUtils.executeIn(null, cloneCommandLine.build());
+
+		if (!Strings.isNullOrEmpty(revision)) {
+			ImmutableList.Builder<String> checkoutCommandLine = ImmutableList.builder();
+			checkoutCommandLine.add("hg", "update", revision);
+			ProcessUtils.executeIn(targetDirectory, checkoutCommandLine.build());
+		}
+	}
+
+	@Override
+	public void update(File targetDirectory, String revision, boolean recursive, boolean mirrored) throws IOException {
+		if (!mirrored) {
+			if (hasUnpublishedChanges(targetDirectory)) {
+				ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "--config", "extensions.rebase=", "pull", "--rebase", "--tool", "internal:fail"));
+			} else {
+				ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "pull", "--update"));
+			}
+
+			if (!Strings.isNullOrEmpty(revision) && !revision.equals(getBranch(targetDirectory))) {
+				ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "checkout", revision));
+			}
+		} else {
+			ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "pull"));
+		}
+	}
+
+	@Override
+	public boolean hasChanges(File targetDirectory) throws IOException {
+		VcsStatus.Builder builder = VcsStatus.builder("");
+		getStatusInternal(targetDirectory, builder);
+		VcsStatus status = builder.build();
+		return status.hasUncommittedChanges() || status.hasUnpublishedChanges();
+	}
+
+	@Override
+	public VcsStatus getStatus(File targetDirectory) throws IOException {
+		final VcsStatus.Builder status = VcsStatus.builder(getRevision(targetDirectory));
+		status.withBranch(getBranch(targetDirectory));
+		getStatusInternal(targetDirectory, status);
+		return status.build();
+	}
+
+	private void getStatusInternal(File targetDirectory, final VcsStatus.Builder status) throws IOException {
+		checkModified(targetDirectory, status);
+		checkOutgoing(targetDirectory, status);
+	}
+
+	private void checkModified(File targetDirectory, final VcsStatus.Builder status) throws IOException {
+		Process process = ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "status"), false, false);
+		String result = new String(ByteStreams.toByteArray(process.getInputStream()), Charsets.UTF_8);
+		status.withUncommittedChanges(!result.isEmpty());
+	}
+
+	private void checkOutgoing(File targetDirectory, final VcsStatus.Builder status) throws IOException {
+		boolean unpublishedChanges = hasUnpublishedChanges(targetDirectory);
+		status.withUnpublishedChanges(unpublishedChanges);
+	}
+
+	private boolean hasUnpublishedChanges(File targetDirectory) throws IOException {
+		try {
+			Process process = ProcessUtils.executeIn(targetDirectory, Arrays.asList("hg", "outgoing", "--quiet", "--template", "{node}\\n"), false, false);
+			String result = new String(ByteStreams.toByteArray(process.getInputStream()), Charsets.UTF_8);
+			return !result.isEmpty();
+		} catch (PrideException e) {
+			// `hg outgoing` exits (stupidly) with status code 1 when there are no outgoing changes
+			return false;
+		}
```

Comment:
We should handle this better. Perhaps refactor `ProcessUtils.executeIn()` in a way that it's able to return the exit value when needed (and handle the error as it does now in other cases). Maybe do that in a separate pull-request first.